web-app/src/components/BookingModal/BookingModalForm/index.js

```diff
@@ -0,0 +1,148 @@
+import React, { Fragment } from 'react';
+import moment from 'moment';
+import { PropTypes as PT } from 'prop-types';
+import container from './container';
+import { Form, Input } from '../../common';
+import {
+  getDurationNotice,
+  calculateDaysNotice,
+} from '../../../utilities/dates';
+import Errorbox from '../../common/Errorbox';
+
+const BookingModalForm = props => {
+  const {
+    isEventBeingUpdated,
+    submitHolidayRequest,
+    updateHolidayRequest,
+    deleteHolidayRequest,
+    formData,
+    formStatus,
+    formIsValid,
+    daysRequested,
+  } = props;
+
+  const createCtas = () => {
+    if (isEventBeingUpdated) {
+      return [
+        {
+          label: 'Update',
+          event: updateHolidayRequest,
+          disabled: !formIsValid,
+        },
+        {
+          label: 'Cancel',
+          event: deleteHolidayRequest,
+          disabled: !formIsValid,
+        },
+      ];
+    } else {
+      return [
+        {
+          label: 'Request',
+          event: submitHolidayRequest,
+          disabled: !formIsValid,
+        },
+      ];
+    }
+  };
+
+  const composeErrorMessage = () => {
+    const { eventStatusId, start } = formData;
+    if (isEventBeingUpdated || eventStatusId == 4 || eventStatusId == 5) {
+      return null;
+    } else {
+      const today = new moment();
+      const fromTodayToStartDateRequested = getDurationNotice(today, start);
+
+      const daysNotice = calculateDaysNotice(daysRequested);
+      let error;
+      if (daysRequested == 0) {
+        error = {
+          message: 'Are you sure you wish to request a day off over a weekend?',
+        };
+      } else {
+        error = {
+          message: `You need to give ${daysNotice} working/business days notice to request ${daysRequested} ${
```

Comment:
Maybe change the wording of this message slightly. From "You need" to "You should" so it doesn't sound like such a hard requirement.